tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_access.c

```diff
@@ -245,6 +325,335 @@ static void test_sub_ext_model(void)
 	PASS();
 }
 
+static void pub_param_set(uint8_t period, uint8_t transmit)
+{
+	struct bt_mesh_cfg_mod_pub pub_params = {
+		.addr = UNICAST_ADDR2,
+		.uuid = NULL,
+		.cred_flag = false,
+		.app_idx = 0,
+		.ttl = 5,
+		.period = period,
+		.transmit = transmit,
+	};
+	uint8_t status;
+	int err;
+
+	err = bt_mesh_cfg_mod_pub_set(0, UNICAST_ADDR1, UNICAST_ADDR1, TEST_MODEL_ID_1,
+				      &pub_params, &status);
+	if (err || status) {
+		FAIL("Mod pub set failed (err %d, status %u)", err, status);
+	}
+}
+
+static void msg4_publish(void)
+{
+	struct bt_mesh_model *model = &models[2];
+
+	bt_mesh_model_msg_init(model->pub->msg, TEST_MESSAGE_OP_4);
+	net_buf_simple_add_u8(model->pub->msg, 1);
+	bt_mesh_model_publish(model);
+}
+
+static void pub_jitter_check(int32_t interval, uint8_t count)
+{
+	int64_t timestamp = k_uptime_get();
+	int32_t jitter = 0;
+	int err;
+
+	for (size_t j = 0; j < count; j++) {
+		/* Every new publication will release semaphore in the update handler and the time
+		 * between two consecutive publications will be measured.
+		 */
+		err = k_sem_take(&publish_sem, K_SECONDS(20));
+		if (err) {
+			FAIL("Send timed out");
+		}
+
+		int32_t time_delta = k_uptime_delta(&timestamp);
+		int32_t pub_delta = llabs(time_delta - interval);
+
+		jitter = MAX(pub_delta, jitter);
+
+		LOG_DBG("Send time: %d delta: %d jitter: %d", (int32_t)timestamp, time_delta,
+			jitter);
+	}
+
+	LOG_INF("Send jitter: %d", jitter);
+	ASSERT_TRUE(jitter <= 10);
+}
+
+static void recv_jitter_check(int32_t interval, uint8_t count)
+{
+	int64_t timestamp;
+	uint32_t jitter = 0;
+	int err;
+
+	/* The measurment starts by the first received message. */
+	err = k_sem_take(&publish_sem, K_SECONDS(20));
+	if (err) {
+		FAIL("Recv timed out");
+	}
+
+	timestamp = k_uptime_get();
+
+	for (size_t j = 0; j < count; j++) {
+		/* Every new received message will release semaphore in the message handler and
+		 * the time between two consecutive publications will be measured.
+		 */
+		err = k_sem_take(&publish_sem, K_SECONDS(20));
+		if (err) {
+			FAIL("Recv timed out");
+		}
+
+		int32_t time_delta = k_uptime_delta(&timestamp);
+		int32_t pub_delta = llabs(time_delta - interval);
+
+		jitter = MAX(pub_delta, jitter);
+
+		LOG_DBG("Recv time: %d delta: %d jitter: %d", (int32_t)timestamp, time_delta,
+			jitter);
+	}
+
+	LOG_INF("Recv jitter: %d", jitter);
+	ASSERT_TRUE(jitter <= 35);
```

Comment:
How did you choose the jitter values for comparisons? Will be jitter worse on PCs with low performance?

Reply:
Roughly it should not be bigger than the advertising event time multiplied by number of network transmissions. If there is not packet loss, it should be received within the first advertising event. I had to increase it a bit due to a packet loss.

I don't think it should depend on PC performance.